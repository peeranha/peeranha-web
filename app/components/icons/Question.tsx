import React from 'react';
import IconComponent, { IconProps } from './IconComponent';

const Question: React.FC<IconProps> = (props): JSX.Element => (
  <IconComponent
    id="question"
    fill="currentColor"
    viewBox="0 0 18 18"
    size={props.size || [18, 18]}
  >
    <g fill="none">
      <circle
        cx="9"
        cy="9"
        r="8.5"
        fill={props.fill || 'none'}
        stroke={props.stroke || '#282828'}
      />
      <path
        stroke="none"
        fill={props.stroke || '#282828'}
        d="M9 4.235c-1.46 0-2.647 1.224-2.647 2.73 0 .309.244.56.544.56.3 0 .544-.251.544-.56 0-.887.7-1.608 1.559-1.608.86 0 1.559.721 1.559 1.607s-.7 1.607-1.56 1.607c-.3 0-.543.251-.543.561v1.954c0 .31.243.561.544.561.3 0 .544-.251.544-.56V9.633c1.2-.259 2.103-1.358 2.103-2.67 0-1.505-1.187-2.729-2.647-2.729M9 12.706a.53.53 0 1 0 0 1.06.53.53 0 0 0 0-1.06"
      />
    </g>
  </IconComponent>
);

export default Question;
