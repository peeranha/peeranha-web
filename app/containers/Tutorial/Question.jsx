import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import ReactPlayer from 'react-player';
import createdHistory from 'createdHistory';

import BaseTransparent from 'components/Base/BaseTransparent';

import Span from 'components/Span';
import ArrowDownFillIcon from 'icons/ArrowDownFill';

import {
  BG_SECONDARY_SPECIAL_4,
  BG_TRANSPARENT,
  BORDER_TRANSPARENT,
  TEXT_PRIMARY,
  TEXT_DARK,
  BORDER_PRIMARY_LIGHT,
} from 'style-constants';
import LoadingIndicator from 'components/LoadingIndicator/CenteredTutorial';

const QuestionBox = BaseTransparent.extend`
  display: flex;
  align-items: baseline;
  padding: 10px 30px;
  background: ${x => (x.isOpened ? BG_SECONDARY_SPECIAL_4 : BG_TRANSPARENT)};
  border: 1px solid
    ${x => (x.isOpened ? BORDER_PRIMARY_LIGHT : BORDER_TRANSPARENT)};

  h5 span {
    color: ${x => (x.isOpened ? TEXT_PRIMARY : TEXT_DARK)};
  }

  &:first-child {
    padding-top: 15px;
  }

  &:last-child {
    padding-bottom: 15px;
  }
`.withComponent('li');

const QuestionBoxBody = styled.div`
  width: 100%;
`;

const ImgWrapper = styled.div`
  margin-right: 18px;
  width: 42px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;

  @media only screen and (max-width: 576px) {
    margin-right: 8px;
  }
`;

const VideoWrapper = styled.div`
  position: relative;
  display: ${x => (x.isOpened ? 'block' : 'none')};
  height: 40vh;
  margin-top: ${x => (x.isOpened ? '15px' : '0px')};
  margin-bottom: 5px;
`;

const getYoutubeLink = mdContent =>
  mdContent.match(/https:\/\/www\.youtube\.com\//)
    ? mdContent.match(/https:\/\/www\.youtube\.com\//).input.split(`"`)[1]
    : null;

const VideoBlock = ({ isOpened, content, sectionIsOpened }) => {
  const [isLoading, setIsLoading] = useState(true);
  const [isPaused, setIsPaused] = useState(true);
  return (
    <VideoWrapper isOpened={isOpened}>
      {isLoading && <LoadingIndicator />}
      <ReactPlayer
        url={getYoutubeLink(content)}
        controls
        playing={isOpened && sectionIsOpened && !isPaused}
        width="100%"
        height="100%"
        onReady={() => setIsLoading(false)}
        onPause={() => setIsPaused(true)}
        onPlay={() => setIsPaused(false)}
      />
    </VideoWrapper>
  );
};

const Question = ({
  h3,
  content,
  questionCode,
  sectionCode,
  route,
  getQuestionCode,
  sectionIsOpened,
}) => {
  const { hash } = window.location;

  const [isOpened, collapse] = useState(false);

  const questionId = getQuestionCode(sectionCode, questionCode);

  const collapseQuestion = () => {
    createdHistory.push(route());
    collapse(!isOpened);
  };

  if (hash.match(questionId) && !isOpened) {
    collapse(true);
  }

  return (
    <QuestionBox id={questionId} isOpened={isOpened}>
      <ImgWrapper>
        <ArrowDownFillIcon className={isOpened && 'transform180'} />
      </ImgWrapper>

      <QuestionBoxBody>
        <h5 className="d-flex align-items-center" onClick={collapseQuestion}>
          <Span fontSize="20" lineHeight="30" mobileFS="16">
            {h3}
          </Span>
        </h5>

        <VideoBlock
          sectionIsOpened={sectionIsOpened}
          isOpened={isOpened}
          content={content}
        />
      </QuestionBoxBody>
    </QuestionBox>
  );
};

Question.propTypes = {
  h3: PropTypes.string,
  content: PropTypes.string,
  questionCode: PropTypes.number,
  sectionCode: PropTypes.number,
  route: PropTypes.func,
  getQuestionCode: PropTypes.func,
  sectionIsOpened: PropTypes.bool,
};

VideoBlock.propTypes = {
  isOpened: PropTypes.bool,
  content: PropTypes.string,
  sectionIsOpened: PropTypes.bool,
};

export default Question;
