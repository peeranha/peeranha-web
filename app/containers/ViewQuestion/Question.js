import React from 'react';
import PropTypes from 'prop-types';

import QuestionTitle from './QuestionTitle';
import Content from './Content';

const Question = props => (
  <div className="question">
    <QuestionTitle title={props.questionData.content.title} />
    <Content
      {...props}
      answerId="0"
      type="question"
      key={`question${props.questionData.id}`}
      comments={props.questionData.comments}
      content={props.questionData.content.content}
      rating={props.questionData.rating}
      isItWrittenByMe={props.questionData.isItWrittenByMe}
      history={props.questionData.history}
      userInfo={props.questionData.userInfo}
      postTime={props.questionData.post_time}
      lastEditedDate={props.questionData.lastEditedDate}
      votingStatus={props.questionData.votingStatus}
      deleteItem={props.deleteQuestion}
      editItem={props.editQuestion}
      editComment={props.editComment}
      saveComment={props.saveComment}
      editCommentState={props.editCommentState}
      deleteComment={props.deleteComment}
      buttonParams={{
        questionId: props.questionData.id,
        answerId: 0,
      }}
    />
  </div>
);

Question.propTypes = {
  questionData: PropTypes.object,
  editCommentState: PropTypes.object,
  deleteQuestion: PropTypes.func,
  editQuestion: PropTypes.func,
  editComment: PropTypes.func,
  saveComment: PropTypes.func,
  deleteComment: PropTypes.func,
};

export default Question;
