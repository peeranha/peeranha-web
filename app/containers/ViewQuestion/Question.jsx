import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import * as routes from 'routes-config';

import Content from './Content';
import { QUESTION_TYPE, POST_TYPES } from './constants';

export const Question = props => {
  const { postType, id } = props.questionData;

  const type = useMemo(
    () => {
      return POST_TYPES[postType];
    },
    [postType],
  );

  return (
    <Content
      {...props}
      answerId={0}
      type={QUESTION_TYPE}
      key={`${QUESTION_TYPE}${props.questionData.id}`}
      comments={props.questionData.comments}
      title={props.questionData.title}
      content={props.questionData.content}
      rating={props.questionData.rating}
      isItWrittenByMe={props.questionData.isItWrittenByMe}
      history={props.questionData.history}
      author={props.questionData.author}
      postTime={props.questionData.postTime}
      lastEditedDate={props.questionData.lastEditedDate}
      votingStatus={props.questionData.votingStatus}
      deleteItem={props.deleteQuestion}
      deleteItemLoading={props.deleteQuestionLoading}
      editItem={[
        props.redirectToEditQuestionPage,
        routes.questionEdit(type, id),
      ]}
      saveComment={props.saveComment}
      deleteComment={props.deleteComment}
      buttonParams={{
        questionId: props.questionData.id,
        answerId: 0,
        whowasvoted: props.questionData.author.id,
      }}
      communities={props.communities}
    />
  );
};

Question.propTypes = {
  questionData: PropTypes.object,
  communities: PropTypes.array,
  deleteQuestion: PropTypes.func,
  saveComment: PropTypes.func,
  redirectToEditQuestionPage: PropTypes.func,
  deleteComment: PropTypes.func,
  deleteQuestionLoading: PropTypes.bool,
};

export default React.memo(Question);
