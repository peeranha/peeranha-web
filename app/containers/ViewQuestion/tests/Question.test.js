import { Question } from '../Question';

describe('Question', () => {
  it('test', () => {
    const props = {
      questionData: {
        userInfo: {},
        content: {
          title: 'Title',
        },
      },
    };

    expect(Question(props)).toMatchSnapshot();
  });
});
