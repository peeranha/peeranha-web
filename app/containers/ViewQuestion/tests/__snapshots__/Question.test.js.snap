// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question test 1`] = `
<div
  className="question"
>
  <QuestionTitle
    title="Title"
  />
  <Content
    answerId="0"
    buttonParams={
      Object {
        "answerId": 0,
        "questionId": undefined,
      }
    }
    questionData={
      Object {
        "content": Object {
          "title": "Title",
        },
      }
    }
    type="question"
  />
</div>
`;
